internal/bsr/convert/convert_test.go

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package convert_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/bsr"
+	"github.com/hashicorp/boundary/internal/bsr/convert"
+	"github.com/hashicorp/boundary/internal/bsr/internal/fstest"
+	"github.com/hashicorp/boundary/internal/bsr/kms"
+	"github.com/hashicorp/boundary/internal/bsr/ssh"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConvert_ToAsciicast(t *testing.T) {
+	ctx := context.Background()
+
+	fs := &fstest.MemFS{}
+	tmpfile, err := fstest.NewTempFile(t.Name())
+	require.NoError(t, err)
+	ts := time.Date(2023, time.March, 16, 10, 47, 3, 14, time.UTC)
+
+	connectionId := "test_connection"
+	channelId := "test_channel"
+
+	writeToChannel := func(w io.Writer, chunks ...bsr.Chunk) {
+		w.Write(bsr.Magic.Bytes())
+		enc, err := bsr.NewChunkEncoder(ctx, w, bsr.NoCompression, bsr.NoEncryption)
+		require.NoError(t, err)
+
+		for _, c := range chunks {
+			_, err := enc.Encode(ctx, c)
+			require.NoError(t, err)
+		}
+	}
+
+	cases := []struct {
```

Comment:
Can you include test cases for all of the [valid ssh.SessionPrograms](https://github.com/hashicorp/boundary/blob/456c5dbb4d6c64886abcacb425b7faf5a39de673/internal/bsr/ssh/types.go#L39-L48). It looks like this already has some for `ssh.Exec`, `ssh.Shell` and `ssh.Sumsystem`, so can you add ones for `ssh.None` and `ssh.NotApplicable`?